tests/panels-features/panels-features-history-panel.spec.js

```diff
@@ -50,3 +56,258 @@ mainTest(
+      });
+
+      mainTest(
+        qase(1938, 'Restore autosaved version via history panel'),
+        async ({ page }) => {
+          const historyPage = new HistoryPanelPage(page);
+          const layersPanelPage = new LayersPanelPage(page);
+          await historyPage.selectSnapshotOption('Restore version');
+          await historyPage.clickRestoreVersionButton();
+          await layersPanelPage.isLayerPresentOnLayersTab('Rectangle', false);
+          await historyPage.isHistoryPanelVisible(false);
+          await historyPage.clickHistoryPanelButton();
+          await historyPage.checkAutosaveVersionsCount('2');
+        },
+      );
+
+      mainTest(
+        qase(1943, 'Pin the autosaved version via history panel'),
+        async ({ page }) => {
+          const versionName = 'pin version';
+          const historyPage = new HistoryPanelPage(page);
+          const layersPanelPage = new LayersPanelPage(page);
+          await historyPage.selectSnapshotOption('Pin version');
+          await historyPage.renameVersion(versionName);
+          await historyPage.checkLastVersionName(versionName);
+          await historyPage.isAutosaveVersionsVisible(false);
+        },
+      );
+    });
+  });
+
+  mainTest(qase(1934, 'Rename version via history panel'), async ({ page }) => {
+    const versionName = 'renamed version';
+    const historyPage = new HistoryPanelPage(page);
+    await historyPage.selectVersionOption('Rename');
+    await historyPage.renameVersion(versionName);
+    await historyPage.checkVersionName(versionName);
+  });
+
+  mainTest(qase(1935, 'Delete version via history panel'), async ({ page }) => {
+    const historyPage = new HistoryPanelPage(page);
+    await historyPage.selectVersionOption('Delete');
+    await historyPage.isVersionListEmpty();
+  });
+});
+
+mainTest.describe(() => {
+  mainTest.afterEach(async ({ page }, testInfo) => {
+    const teamPage = new TeamPage(page);
+    const mainPage = new MainPage(page);
+    const profilePage = new ProfilePage(page);
+    const loginPage = new LoginPage(page);
+    const dashboardPage = new DashboardPage(page);
+    await mainPage.backToDashboardFromFileEditor();
+    await profilePage.logout();
+    await loginPage.waitLoginPage();
+    await loginPage.isLoginPageOpened();
+    await loginPage.enterEmail(process.env.LOGIN_EMAIL);
+    await loginPage.enterPwd(process.env.LOGIN_PWD);
+    await loginPage.clickLoginButton();
+    await dashboardPage.isDashboardOpenedAfterLogin();
+    await teamPage.switchTeam(teamName);
+    await dashboardPage.openFile();
+  });
+
+  mainTest(
+    qase(
+      [1939, 1940, 1941],
+      "Verification of displaying other users' versions in the version list" +
+        'Setting "My Versions/ All Versions" filters in the version list',
+    ),
+    async ({ page }, testInfo) => {
+      const versionName = 'test version';
+      await testInfo.setTimeout(testInfo.timeout + 30000);
```

Comment:
Could you implement the test.slow() solution for these tests ([playwright doc](https://playwright.dev/docs/api/class-test#test-slow))?

Reply:
done